src/drawer/index.tsx

```diff
@@ -30,30 +28,23 @@ export default class Drawer extends PureComponent <DrawerProps, DrawerState> {
     width: '400px',
     onOpen: () => {},
     onClose: () => {},
-    onChange: (isOpen) => {},
-  }
-
-  state: DrawerState = {
+    onTransitionEnd: (isOpen) => {},
     open: false,
   }
 
   componentDidMount() {
     document.addEventListener('mouseup', this.handleOutsideMouseClick)
     document.addEventListener('touchstart', this.handleOutsideMouseClick)
     document.addEventListener('keydown', this.handleKeydown)
-
-    // Apply "open" CSS animation using requestAnimationFrame as browsers are not rerendering
-    // stuff that changed in the same animation frame.
-    requestAnimationFrame(() => {
-      // This nested requestAnimationFrame here is only needed for Firefox
-      requestAnimationFrame(this.open)
-    })
   }
 
-  componentWillReceiveProps({ width }: DrawerProps) {
+  componentDidUpdate({ width, open }: DrawerProps) {
     if (this.props.width !== width) {
```

Comment:
It won't work anymore. This was done before the next render. Here, if you don't have any render after the componentDidUpdate, you won't see any update of the this.contentStyles.width.

Reply:
Moved to `render`